cou/utils/juju_utils.py

```diff
@@ -88,17 +84,19 @@ async def _async_get_model(model_name: Optional[str] = None) -> Model:
     :param model_name: the juju.model.Model object to fetch
     :type model_name: Optional[str]
     :returns: juju.model.Model
+    :rtype: Model
     """
     global CURRENT_MODEL
 
     model = CURRENT_MODEL
     if model is not None and _is_model_disconnected(model):
         await _disconnect(model)
-        model = None
-    if CURRENT_MODEL is None:
+
+    if model is None:
```

Comment:
A disconnected model object is not `None`, so we are not able to enter this block after executing L93. We may unintentionally return a disconnected model at the end of this function

Reply:
To be honest I'm confused from this function completely. I'm reverting it back and I will reworked it in next PR where we use Model object instead of model_name pass to all other functions. Is that ok for you?

Reply:
Yes, that sounds great. Thanks.